src/umocktypes_wcharptr.c

```diff
@@ -36,7 +36,12 @@ char* umocktypes_stringify_wcharptr(const wchar_t** value)
         {
             /* Codes_SRS_UMOCKTYPES_WCHARPTR_01_002: [ umocktypes_stringify_wcharptr shall return a string containing the string representation of value, enclosed by quotes ("value"). ] */
             const wchar_t* same_as_source = *value;
-            mbstate_t state = { 0 };
+            
+            #if defined (_clang_)
+                mbstate_t state = { {0} };
+            #else
```

Comment:
Ok. Spent some time looking at this. It's not right to initialize the structure's fields with 0, because the standard does not actually specify what fields are there.

The proper way to clear this mbstate_t is to actually call
wcsrtombs with a pointer to an empty string ("") as source.
In that case, per the standard the ps argument will be set to the initial state:

C99 Section 7.24.6.4.2 The wcsrtombs function

...

Paragraph 3:

"If conversion stopped due to
reaching a terminating null wide character, the resulting state described is the initial
conversion state."

What can and probably will happen is that another toolset will have a slightly different mbstate_t and then we have to chain yet another define. So, instead of that let's do this:
Let's have a static function in this file

void mbstate_t_init(mbstate_t*)

which should call wcsrtombs with an empty string ("") as source, like here (scroll down to the bottom example):
[https://www.gnu.org/software/libc/manual/html_node/Keeping-the-state.html](https://www.gnu.org/software/libc/manual/html_node/Keeping-the-state.html)